src/Simplex/Messaging/Agent/Protocol.hs

```diff
@@ -1170,7 +1202,7 @@ data SMPQueueInfo = SMPQueueInfo {clientVersion :: VersionSMPC, queueAddress ::
 
 instance Encoding SMPQueueInfo where
   smpEncode (SMPQueueInfo clientVersion SMPQueueAddress {smpServer, senderId, dhPublicKey, queueMode})
-    | clientVersion >= shortLinksSMPClientVersion = addrEnc <> smpEncode queueMode
+    | clientVersion >= shortLinksSMPClientVersion = addrEnc <> maybe "" smpEncode queueMode
```

Comment:
this is indeed to avoid potential conflict/overlap with subsequent Maybe in ConnectionRequestUri encoding.